### PR TITLE
My Home: Update celebratory notices (take 2)

### LINKED ISSUE #41998: My Home: Celebratory Cards in new Layout
As proposed by @danhauk, to better match the focus of one thing at a time on My Home, let's update the celebratory cards to render in the primary layout slot. **Note that certain design and flow details are still being worked through and are open to feedback.**

after

Each celebratory card will also get a different and ✨ illustration thanks to @danhauk 
illustrations
(Folks are still working on the migration illustration).

### Site Creation
80830182-892da480-8bb6-11ea-85c8-f5ed32f19d0a

* Clicking on "Get Started" immediately transitions into the site setup list (with some sort of animation).
* This banner is only visible on first page view after site creation by the site creator, any other navigation to this section or page refresh should transition next to the site-setup list.

### Launch Site after checklist complete
80830606-3d2f2f80-8bb7-11ea-9c9a-0c007898d15f

* Show me what's next would immediately transition to the next task
* "Dismiss" would dismiss the card and the next task would show up after the waiting period (same as the x in the existing banners)

### Site launched before checklist complete
80830961-ef66f700-8bb7-11ea-9dab-e13bc6d3c7b1

* "Show site setup" shows the site setup list again
* "Skip site setup" marks the whole checklist as complete and dismisses the card. The next task would show up after the waiting period.
* How long is this visible for without a user click (is there an auto time expiry?)

### Site setup complete after launch
80831081-25a47680-8bb8-11ea-8197-76386e340fc2

* Show me what's next would immediately transition to the next task
* "Dismiss" would dismiss the card and the next task would show up after the waiting period (same as the x in the existing banners)

### Migration 
* Show me what's next would immediately transition to the next task
* "Dismiss" would dismiss the card and the next task would show up after the waiting period (same as the x in the existing banners)

### PR DESCRIPTION
**Pull request by kwight**
### Changes proposed in this Pull Request

Adds a new version for the celebratory notices so they fit in better with the new layout.

May-13-2020 15-53-50

**Site created**
Before | After
--- | --- 
 image | image

**Site launched before site setup completed**
Before | After
--- | --- 
image | image

**Site launched after site setup completed**
Before | After
--- | --- 
image | image

**Site setup complete after site launched**
Before | After
--- | --- 
 image | image

**Site migrated**
Before | After
--- | --- 
image | image

### Testing instructions

* Apply D43488-code and sandbox the API.

#### Flow 1: Launch before completing site setup
* Create a new site.
* Make sure the site created notice shows up.
* Click on "Get started".
* Make sure the notice is replaced with the site setup list.
* Launch the site.
* Make sure the site launched notice shows up.
* Click on "Show site setup".
* Make sure the notice is replaced with the site setup list.
* Complete the site setup.
* Make sure the site setup complete notice shows up.
* Click on "Show me what's next".
* Make sure the notice is replaced with the webinars task.

#### Flow 2: Launch after completing site setup
* Create one more site and complete the site setup before launching this time.
* Make sure the site launched notice shows up.
* Click on "Show me what's next".
* Make sure the notice is replaced with the webinars task.

#### Flow 3: Skip site setup
* Create one more site and launch it before completing the site setup (like in flow 1).
* Click on "Skip site setup" this time.
* Make sure the launch notice is dismissed and it is not replaced by any task.
* Reload My Home and verify the webinars task shows up.

#### Flow 4: Dismiss site setup complete celebratory notice
* Create one more site and launch it before completing the site setup (like in flow 1).
* Click on "Show site setup" and complete the site setup.
* Click on "Dismiss" this time.
* Make sure the site setup notice is dismissed and it is not replaced by any task.
* Reload My Home and verify the webinars task shows up.

#### Flow 5: Dismiss site launched celebratory notice
* Create one more site and complete the site setup before launching (like in flow 2).
* Make sure the site launched notice shows up.
* Click on "Dismiss" this time.
* Make sure the site launched is dismissed and it is not replaced by any task.
* Reload My Home and verify the webinars task shows up.

#### Flow 6: Import a site
* Go to Tools > Import > WordPress.
* Enter the URL of a Jetpack site and select Everything.
* Proceed with the import.
* Make sure the site migrated notice shows up.
* Click on "Show me what's next".
* Make sure the notice is replaced with the webinars task.

Fixes https://github.com/Automattic/wp-calypso/issues/41998  https://github.com/Automattic/dotcom-private/issues/50

### CODE REVIEW

File: client/my-sites/customer-home/cards/notices/celebrate-notice-v2/index.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	if ( ! isVisible ) {
+		return null;
+	}
+
+	const dismissalPreferenceKey = `dismissible-card-home-notice-${ noticeId }-${ siteId }`;
+
+	const showNextTask = () => {
+		setIsLoading( true );
+		dispatch( savePreference( dismissalPreferenceKey, true ) );
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/notices/celebrate-notice-v2/index.jsx:51]: So here's the race condition, we can request the next layout fetch before this is updated on the backend.

File: client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx [hunk 2]
```diff
@@ -61,6 +61,7 @@ const startTask = ( dispatch, task, siteId ) => {
 
 const skipTask = ( dispatch, task, siteId ) => {
 	dispatch( requestSiteChecklistTaskUpdate( siteId, task.id ) );
+	dispatch( requestHomeLayout( siteId ) );
```
Comment (gwwar) [inline client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx:64]: There's likely a similar race condition here but we can fix/bulletproof in a follow up

File: client/my-sites/customer-home/cards/notices/celebrate-notice-v2/index.jsx (hunk 1 above)
Comment (gwwar) [reply client/my-sites/customer-home/cards/notices/celebrate-notice-v2/index.jsx:51]: I can fix this on the client, but in the future though it'd be more robust to create another POST endpoint to do both parts in one request. Maybe something off of `/home/skip-task`

**Review by gwwar — APPROVED** [review]
Giving tentative approval here since I'm pretty sure cc55ebb will fix the race condition with a stuck spinner. The only other piece to verify is any updates to E2E tests before another landing attempt.

@mmtr @kwight let me know what you think about long term preferences for follow up
- A new POST endpoint to make sure task updates/layout fetch happens in one request
- Keep the client a bit more complex but refactor preferences actions to pure actions + add a general callback option to dispatch a serialized action on API success
- No-op, or keep middleware (I suspect this will be a bit harder to control imo)

**Comment by kwight** [comment]
@gwwar your race condition commit seems to work really well for me ✨

**Comment by kwight** [comment]
> let me know what you think about long term preferences for follow up

I admit I like the first option the most, but it's also the one that seems simplest and easy for me to understand.

**Comment by gwwar** [comment]
@bsessions85 @mmtr 7c8d7d1 should keep E2Es green during the landing attempt. Basically I've updated the signup step to dismiss the celebration card, then check if the site setup list exists. If it's too flaky we can maybe reduce that to verifying that the celebration card is there

> I admit I like the first option the most

@kwight another endpoint would be my first preference too since it's the simplest and also fastest for the user. (We only need to wait for one request vs two). We can see what @mmtr  and Ajax thinks though

**Comment by gwwar** [comment]
Note that e2es will not be green until the server side patch lands

**Comment by mmtr** [comment]
> > let me know what you think about long term preferences for follow up
> 
> I admit I like the first option the most, but it's also the one that seems simplest and easy for me to understand.

I also like this and kinda aligns with our API-powered My Home approach.

**Comment by mmtr** [comment]
> Note that e2es will not be green until the server side patch lands, so they should be deployed at roughly the same time.

I just pushed a a few commit that tries to close both celebratory notices, so they should pass now both before and after landing the server-side patch.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3585651

Thank you @kwight for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**